libs/scalalib/spotless-worker/src/mill/scalalib/spotless/GitRatchet.scala

```diff
@@ -0,0 +1,73 @@
+package mill.scalalib.spotless
+
+import mill.define.{PathRef, TaskCtx}
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.dircache.DirCacheIterator
+import org.eclipse.jgit.lib.{RepositoryBuilder, RepositoryCache}
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.filter.{AndTreeFilter, PathFilter, TreeFilter}
+import org.eclipse.jgit.treewalk.{CanonicalTreeParser, FileTreeIterator}
+import org.eclipse.jgit.util.FS
+
+import scala.jdk.CollectionConverters.*
+import scala.util.Using
+
+/**
+ * @see [[https://github.com/centic9/jgit-cookbook]]
+ */
+class GitRatchet(root: os.Path) extends AutoCloseable {
+
+  val repository = RepositoryBuilder().setFS(FS.DETECTED).findGitDir(root.toIO).build()
+
+  // HACK:
+  // Forcibly load classes referenced in close to avoid: java.lang.NoClassDefFoundError.
```

Comment:
IIUC, the error is thrown because `close` is not called with the worker classloader.